site/src/components/RequirePermission/RequirePermission.tsx

```diff
@@ -0,0 +1,18 @@
+import { FC } from "react"
+import { Navigate } from "react-router"
+
+export interface RequirePermissionProps {
+  children: JSX.Element
```

Comment:
Would [`PropsWithChildren`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L799) help here? [reference](https://www.mendelowski.com/docs/react/children-props-with-typescript/)

Reply:
I tried using it here and got a TS error, not sure why!

Reply:
This works in the playground: https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDGMA0cDecAKUEYAzgOrAwAWAwlcADYAmUSAdnAL5wBmRIcAOSt0MQQFgAUFKQAPSLDgwAnmCSIkARwCuwVviRQQwEiWAQ2hYiTgBeAkVIVqdRi3YAeHFLhwTAMVQYbVYANRNgACMGJAAuOEiICBiUNilOAD4pKTQLEngAORQAN2AAcxQYdXsACjBHEnicGAh4-KhgNjLOAEo7DLgPFDgqVh5bHHrrADoWzIZOgGsPAHoULOlJFYAqbZ9tuDIoFFI4Im0qm2pKuBRWW+KURhRo9UiUEiQmOAtEACEAII0H5QOALNDsMxdab7FYyeTQeC5Nj5DQ6PRIAxGExmX61PBoejMVhsbABIIhJDhMyvLjxZDo-SGYymcyWBp9WwDbySXzAHhwGoAQnJlUp1KiMT6PN8vlYwSgHA8RVKFSqSggtgARCsAO7QRYkMDoJAkLVwFYbXzcJAMT64Hyy+UhDiEtwkx2cdJAA

Up to you though: just a suggestion